### DISCUSSION TITLE
What happens when code changes between task scheduling and execution?

### DISCUSSION THREAD

**Discussion by daniboygg** [RealOrangeOne/django-tasks#110] [Q&A] [answered]
So I was wondering what happens in the following case using the database backend. 

1. A task called `task_1` is enqueued for later execution and inserted into the database.
2. The code for `task_1` is modified.
3. The updated code is deployed, causing the server and the database worker to restart with the new code.
4. The task is now scheduled to run with code different from what was originally enqueued.

I have searched in the code, but I don't see anything to add extra data/context to the task. 
Could be interesting to add the version of the code that enqueued the task, so when is running let the user decide if this task should execute or cancelling?

**Comment by RealOrangeOne** [comment, marked as answer]
Currently, it's whatever is deployed at the time the task is **executed** that will be run - not whatever is running at enqueue time. In most cases, this is probably fine. For tasks using `run_after`, this could be a problem, but I think it's fairly niche.

At some point, I'd like to add a TTL to tasks, whereby if they're unexecuted for too long, they're cancelled. That might be a way to work around this if it were causing issues. 

I think the ideal solution is to ensure task changes are backwards compatible for applications with large queue depths.
